src/dashboard/apps/consent/views.py

```diff
@@ -2,102 +2,108 @@
 
 from django.contrib import messages
 from django.core.exceptions import PermissionDenied
-from django.db.models import TextField
-from django.db.models.functions import Cast
-from django.shortcuts import get_object_or_404, redirect, render
+from django.db.models import Q
+from django.shortcuts import get_object_or_404, redirect
 from django.urls import reverse_lazy as reverse
 from django.utils import timezone
 from django.utils.translation import gettext_lazy as _
-from django.views.generic import TemplateView
+from django.views.generic import FormView, TemplateView
 
 from apps.core.models import Entity
 
+from ..auth.models import DashboardUser
 from . import AWAITING, VALIDATED
+from .forms import ConsentForm
+from .mixins import BreadcrumbContextMixin
 from .models import Consent
 
 
-class IndexView(TemplateView):
+class IndexView(BreadcrumbContextMixin, TemplateView):
     """Index view of the consent app."""
 
     template_name = "consent/index.html"
+    breadcrumb_current = _("Consent")
 
-    def get_context_data(self, **kwargs):
-        """Add custom context to the view."""
+    def get_context_data(self, **kwargs):  # noqa: D102
         context = super().get_context_data(**kwargs)
-        context["entities"] = self.request.user.get_entities()
+        user: DashboardUser = self.request.user  # type: ignore
+        context["entities"] = user.get_entities()
```

Comment:
Is this required as `request.user` is always in the context, and, `request.user.get_entities` can also be called from a template?

Reply:
`get_entities()` was not recognized by my code analyzer with `self.request.user.get_entities()`, but the problem actually came from my IDE, it's fixed